packages/opentelemetry-exporter/src/index.js

```diff
@@ -0,0 +1,288 @@
+const { ExportResultCode } = require('@opentelemetry/core');
+const instanaEndpointUrlEnvVar = 'INSTANA_ENDPOINT_URL';
+const instanaAgentKeyEnvVar = 'INSTANA_AGENT_KEY';
+
+/** @typedef {import('@opentelemetry/sdk-trace-base').ReadableSpan} ReadableSpan */
+/** @typedef {import('@instana/core/src/tracing/cls').InstanaBaseSpan} InstanaBaseSpan */
+
+// The exporter is based on the parser from the Agent:
+// eslint-disable-next-line max-len
+// https://github.com/instana/agent/blob/51da097e99e9d05ac5c0b5b7dd9275a12c20685d/agent-optional-parent/agent-open-telemetry/src/main/java/com/instana/agent/main/impl/otel/OpenTelemetryTraceEndpoint.java#L69
+
+/** Span Kind
+  Default value. Indicates that the span is used internally.
+  INTERNAL = 0,
+
+  Indicates that the span covers server-side handling of an RPC or other
+  remote request.
+  SERVER = 1,
+
+  Indicates that the span covers the client-side wrapper around an RPC or
+  other remote request.
+  CLIENT = 2,
+
+  Indicates that the span describes producer sending a message to a
+  broker. Unlike client and server, there is no direct critical path latency
+  relationship between producer and consumer spans.
+  PRODUCER = 3,
+
+  Indicates that the span describes consumer receiving a message from a
+  broker. Unlike client and server, there is no direct critical path latency
+  relationship between producer and consumer spans.
+  CONSUMER = 4
+*/
+const SPAN_KIND_INTERNAL = 0;
+const SPAN_KIND_SERVER = 1;
+const SPAN_KIND_CLIENT = 2;
+const SPAN_KIND_PRODUCER = 3;
+const SPAN_KIND_CONSUMER = 4;
+const SERVER_KIND = 'server';
+const CLIENT_KIND = 'client';
+const PRODUCER_KIND = 'producer';
+const CONSUMER_KIND = 'consumer';
+const INTERNAL_KIND = 'internal';
+/**
+ * https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto#L314
+ * Status Code
+ *
+ * Default
+ * UNSET = 0,
+ * The operation has been validated by an Application developer or
+ * Operator to have completed successfully.
+ * OK = 1,
+ * The operation contains an error.
+ * ERROR = 2
+ */
+
+const DEPRECATED_STATUS_CODE_OK = 0;
+const STATUS_UNSET = 0;
+const STATUS_OK = 1;
+// const STATUS_ERROR = 2;
+
+class InstanaExporter {
+  /**
+   * @param {{ agentKey?: string, endpointUrl?: string }}
+   */
+  constructor({ agentKey, endpointUrl }) {
+    this._isShutdown = false;
+    // If endpoint URL and agent key are not provided, we don't set them in an attempt to use env vars, if they were set
+    if (endpointUrl) {
+      process.env[instanaEndpointUrlEnvVar] = endpointUrl;
+    }
+    if (agentKey) {
+      process.env[instanaAgentKeyEnvVar] = agentKey;
+    }
+
+    if (!process.env[instanaEndpointUrlEnvVar] || !process.env[instanaAgentKeyEnvVar]) {
+      throw new Error('The Instana endpoint URL and/or the agent key were not provided');
+    }
+
+    instanaEnvironment.validate();
+    instanaBackendConnector.init();
```

Comment:
_question:_ Do we want to pass in an `identityProvider` here to set a different `Host` header than `nodejs-serverless`?

Reply:
As per our talk on Slack, this is not a deal breaker, so we can discuss this further on.
Will keep as is for now :)